k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSettings.java

```diff
@@ -703,12 +710,21 @@ public boolean onPreferenceClick(Preference preference) {
             mCryptoSupportSignOnly = (CheckBoxPreference) findPreference(PREFERENCE_CRYPTO_SUPPORT_SIGN_ONLY);
 
             mCryptoApp.setValue(String.valueOf(mAccount.getCryptoApp()));
+            if (OpenPgpAppPreference.isApgInstalled(getApplicationContext())) {
+                mCryptoApp.addLegacyProvider(APG_PROVIDER_PLACEHOLDER, getString(R.string.apg), R.drawable.ic_apg_small);
```

Comment:
Internally, the whole item is just a placeholder and not loaded from the APG package at all. But since I only do that when APG is installed I guess we could just as well load the icon from the package...